infra/cifuzz/fuzz_target.py

```diff
@@ -149,25 +148,28 @@ def fuzz(self):
     # We found a bug but we won't report it.
     return FuzzResult(None, None, self.latest_corpus_path)
 
-  def free_disk_if_needed(self):
+  def free_disk_if_needed(self, delete_fuzz_target=True):
     """Deletes things that are no longer needed from fuzzing this fuzz target to
     save disk space if needed."""
     if not self.config.low_disk_space:
+      logging.info('Not freeing disk space after running fuzz target.')
       return
-    logging.info(
-        'Deleting corpus, seed corpus and fuzz target of %s to save disk.',
-        self.target_name)
+    logging.info('Deleting corpus and seed corpus of %s to save disk.',
+                 self.target_name)
 
     # Delete the seed corpus, corpus, and fuzz target.
     if self.latest_corpus_path and os.path.exists(self.latest_corpus_path):
       # Use ignore_errors=True to fix
       # https://github.com/google/oss-fuzz/issues/5383.
       shutil.rmtree(self.latest_corpus_path, ignore_errors=True)
 
-    os.remove(self.target_path)
     target_seed_corpus_path = self.target_path + '_seed_corpus.zip'
     if os.path.exists(target_seed_corpus_path):
       os.remove(target_seed_corpus_path)
+
+    if delete_fuzz_target:
```

Comment:
why do we need to add this? were we running into more disk space issues?

Reply:
My questions was more: was the disk so constrained that we were running into disk space issues without this part?